spec/std/concurrent/select_spec.cr

```diff
@@ -100,4 +110,87 @@ describe "select" do
     sleep
     x.should eq 1
   end
+
+  it "select same channel multiple times" do
+    ch = Channel(Int32).new
+
+    spawn do
+      ch.send(123)
+    end
+
+    select
+    when ch.send(456)
+    when x = ch.receive
+    end
+
+    x.should eq 123
+  end
+
+  it "priorize by order when entering in a select" do
+    ch1 = Channel(Int32).new(5)
+    ch2 = Channel(Int32).new(5)
+
+    2.times { ch1.send 1 }
+    2.times { ch2.send 2 }
+
+    select
+    when x = ch1.receive
+    when x = ch2.receive
+    end
+    x.should eq 1
+
+    select
+    when x = ch2.receive
+    when x = ch1.receive
+    end
+    x.should eq 2
+  end
+
+  it "stress select with send/receive in multiple fibers" do
```

Comment:
Does this spec belong here? I can't see `select` used, but maybe the methods called here are an implementation detail for that?